### PR TITLE
feat: struct support

### PR DESCRIPTION
**Pull request by 0xGorilla**
Closes BES-210, BES-211, BES-212

### CODE REVIEW

**Comment by 0xGorilla** [comment]
We depend on the issue mentioned on https://github.com/Consensys/solc-typed-ast/issues/243

File: src/processor.ts [hunk 1]
```diff
@@ -8,38 +8,42 @@ interface IWarning {
...
-        const fileContracts = (source as any).ast.nodes.filter((node: any) => node.nodeType === 'ContractDefinition');
-        fileContracts.forEach((contract: any) => {
-            const nodes = contract.nodes as SolcContractNode[];
-    
-            nodes
-            .filter(node => !ignoredNodeTypes.includes(node.nodeType))
+    sourceUnits.forEach(sourceUnit => {
+        sourceUnit.vContracts.forEach(contract => {
+            [
+                ...contract.vEnums,
```
Comment (gas1cent) [inline src/processor.ts:17]: Removed `vConstructor` from the list because it's included in `vFunctions`

File: src/validator.ts [hunk 2]
```diff
@@ -13,33 +14,62 @@ export function validate(contractNode: NodeToProcess, natspec: Natspec): string[
...
-    // Make sure all defined function parameters have natspec
-    for(let paramName of functionParameters) {
-        if(!natspecParameters.includes(paramName)) {
-            alerts.push(`@param ${paramName} is missing`);
+        // Make sure there is no natspec defined for non-existing returns
+        for(let paramName of natspecReturns) {
+            if(paramName && !functionReturns.includes(paramName)) {
+                alerts.push(`Missing named return for: @return ${paramName}`);
+            }
+        }
```
Comment (0xGorilla) [inline src/validator.ts:42]: can we wrap this in functions to keep the code cleaner? one fn could be `validateReturnParameters`

Comment (0xGorilla) [reply src/validator.ts:42]: same with the StructDefinition

**Review by gas1cent — APPROVED** [review]

**Pull request merged**